acceptance/framework/config/config.go

```diff
@@ -33,14 +34,18 @@ type TestConfig struct {
 	SecondaryKubeContext   string
 	SecondaryKubeNamespace string
 
+	AppNamespace          string
```

Comment:
> For CNINamespace, we made the design decision that CNI was always installed alongside consul. Othere CNI plugins do it differently and install their plugins in their own namespace. I don't think OpenShift needs the CNI to be installed somewhere else to work?

Right. But if someone wants to run Consul in a namespace with restricted PSA enforcement, then they can't run the CNI in that same namespace because the CNI requires privilege. Eventually, (we assume) restricted PSA enforcement is going to be the OpenShift default, so the goal here was to enable testing with that potential future configuration.

> I'd make AppNamespace/SecondaryAppNamespace test specific or OpenShift specific if needed.

This is still relevant to vanilla K8s because you can enable restricted PSA enforcement there as well (although it likely will never be the default like on OpenShift). So I didn't want to make the app namespace settings OpenShift-specific, because it's useful and (and easier) to be able to test this same configuration on vanilla K8s as well.

Reply:
I replaced all these flags with `-restricted-psa-enforcement-enabled` (default false). When true:

1. The Consul namespace(s) passed to the tests via `-kube-namespaces` need to be configured with the restricted PSA enforcement label.
2. The CNI is deployed into the `kube-system` namespace, which is privileged and already exists (so I removed the -cni-namespace flag)
3. I removed the app namespace flags. Instead, ConnectHelper supports a `UseAppNamespace` setting that tests can optionally use (based on whether `-restricted-psa-enforcement-enabled` is set)

Let me know what you think.